src/xSTIR/pSTIR/STIR.py

```diff
@@ -3040,21 +3144,29 @@ def set_relaxation_parameter(self, value):
             self.handle, self.name, 'relaxation_parameter', value)
 
 
-def make_Poisson_loglikelihood(acq_data, likelihood_type='LinearModelForMean',
+#def make_Poisson_loglikelihood(acq_data, likelihood_type='LinearModelForMean',
+def make_Poisson_loglikelihood(acq_data=None, likelihood_type=None,
                                acq_model=None):
     """Makes Poisson loglikelihood.
 
     Selects the objective function based on the acquisition data and likelihood
     model types.
     """
     # only this objective function is implemented for now
-    if likelihood_type == 'LinearModelForMean':
+    #if likelihood_type == 'LinearModelForMean':
+    if likelihood_type is None:
         obj_fun = PoissonLogLikelihoodWithLinearModelForMeanAndProjData()
-        obj_fun.set_acquisition_data(acq_data)
+        if acq_data is not None:
+            obj_fun.set_acquisition_data(acq_data)
```

Comment:
not sure why we're changing this code here. Do we really need to? What happens if `acq_data` is `None`? It'd presumably crash. (Only the listmode case can handle this)

Reply:
@KrisThielemans so far as I can see, Nikos' list mode objective function does not need acquisition data, so `acq_data` must be optional. We need to discuss this with Nikos.

Reply:
Agreed. @NikEfth's listmode function can either use listmode data or the corresponding cache.

However, these lines are for the "old" case with projection data, which do need it.

Reply:
ok, will make this function throw in the non-listmode case

Reply:
presumably the old code did already, but if not, best to put it in, agreed.

I still think we can revert the default to what it as, such that we have minimal code change, but as you prefer.